### PR TITLE
Fix some quality issues, add DeepSource config file

### PR DESCRIPTION
**Pull request by sanketsaurav**
to: @airbnb/streamalert-maintainers
related to: code quality

## Background
[DeepSource](https://deepsource.io) is a **free for open-source** tool that helps developers and teams write good code. It continuously analyzes code changes on every PR and gives a central dashboard to see code health in terms of issues and important metrics. DeepSource is used by teams at [Uber](https://deepsource.io/gh/uber/ludwig) and [Stripe](https://deepsource.io/gh/stripe/stripe-python), among others.

I've added a customized `.deepsource.toml` file for this repo, and made some fixes for issues existing in the code highlighted by DeepSource. To keep analyzing code on every change, integration is easy:

1. **Merge this PR.**
2. **Sign up on DeepSource** and grant access to this repository [here](https://deepsource.io/signup/).
3. **Activate analysis** for this repo [here](http://deepsource.io/gh/airbnb/streamalert).

You can also take a look at the [docs](https://deepsource.io/docs/guides/quickstart.html). I'll be happy to answer any questions! :)

## Changes
 - Fix `PYL-C1801`: Do not use `len()` to check for empty sequence
 - Fix `PYL-C0201`: Use `iteritems()` to iterate dictionary when both key and value are used
 - Fix `FLK8-E713`: Test for membership should be `not in`

-- Sanket.

### CODE REVIEW

File: publishers/community/slack/slack_layout.py [hunk 1]
```diff
@@ -125,8 +125,8 @@ def publish(self, alert, publication):
         publication['@slack.attachments'].append({
             'color': self._color(),
             'fields': [
-                {'title': key.capitalize(), 'value': rule_presentation['fields'][key]}
-                for key in rule_presentation['fields'].keys()
+                {'title': key.capitalize(), 'value': value}
+                for key, value in rule_presentation['fields'].items()
```
Comment (ryandeivert) [inline publishers/community/slack/slack_layout.py]: this is python2 so please use `.iteritems()` here. in python2, `.items()` creates copies of the key/values as tuples in memory and is more costly.

Comment (sanketsaurav) [reply publishers/community/slack/slack_layout.py]: Got it. Pushing a fix!

**Comment by sanketsaurav** [comment]
@ryandeivert Made changes to use `.iteritems()` instead.

**Comment by ryandeivert** [comment]
thanks! @Ryxias would you mind taking a peek at this since it's mostly publisher code changes?

**Comment by Ryxias** [comment]
Closing zombie'd PR

**Pull request closed without merging**